src/main/java/org/ossgang/commons/mapbackeds/Mapbackeds.java

```diff
@@ -157,23 +245,40 @@ public M build() {
     }
 
     public static Set<Method> fieldMethods(Class<?> intfc) {
+        return collectInterfaceMethods(intfc, m -> !m.isDefault());
+    }
+
+    public static Optional<Method> toStringMethod(Class<?> intfc) {
+        Set<Method> toStringMethods = toStringMethods(intfc);
+        if (toStringMethods.size() > 1) {
+            throw new IllegalArgumentException(
+                    "More than one method with a ToString annotation found in the hierarchy!");
+        }
```

Comment:
Should we also enforce at this place that the method signature matches `toString()` - meaning it must return `String` and it must take 0 arguments?

Otherwise we may get some InvocationTargetException when it is first used ...

Reply:
I think this is properly convered:
1) The zero argument is checked when searching
2) When calling it, it is wrapped into an additional Objects.toString(..) ... so you could even put the annotation onto a method returning an int for example ...(as done in the test)

Reply:
OK, I missed that it's anyway only searching 0 argument methods - so yes, it's covered.